src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java

```diff
@@ -190,4 +206,48 @@ private void trySetupSourceRemapping() {
 			}
 		});
 	}
+
+	/**
+	 * Configuration-cache-compatible action for nesting jars.
+	 * Uses a provider to avoid capturing task references at configuration time.
+	 * Do NOT turn me into a record!
+	 */
+	private static class NestJarsAction implements Action<Task>, Serializable {
+		private final Provider<Directory> nestedJarsDir;
+
+		NestJarsAction(Provider<Directory> nestedJarsDir) {
+			this.nestedJarsDir = nestedJarsDir;
+		}
+
+		@Override
+		public void execute(@NotNull Task t) {
+			final Jar jarTask = (Jar) t;
+
+			if (!jarTask.getArchiveFile().isPresent()) {
+				jarTask.getLogger().warn("Jar task archive file not available, skipping JIJ");
```

Comment:
I don't think this is reachable in practice as the task shouldn't be running without its output file configured.

Reply:
true, will remove that unnecessary check